File: source/downloads/code/LeetCode/Python/Add-Two-Numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        addon=0
        ret = ListNode(-1)
        node = ret
        while l1 or l2:
            newval = 0
            if l1 and l2:
                newval = l1.val + l2.val + addon
            elif l1:
                newval = l1.val + addon
            else:
                newval = l2.val + addon
            newNode = ListNode(newval%10)
            node.next = newNode
            node = newNode
            addon = newval / 10
            if l1:
                l1 = l1.next
            if l2:
                l2 = l2.next
        if addon == 1:
            node.next = ListNode(1)
        return ret.next

File: source/downloads/code/LeetCode/Java/Add-Two-Numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode ret = new ListNode(-1);
        ListNode tmp = ret;
        int sum = 0;
        
        while(l1!=null || l2!=null){
            sum /= 10;
            if(l1 != null){
                sum += l1.val;
                l1 = l1.next;
            }
            if(l2 != null){
                sum += l2.val;
                l2 = l2.next;
            }
            tmp.next = new ListNode(sum%10);
            tmp = tmp.next;
        
        } 
        if(sum >=10)
            tmp.next = new ListNode(1);
        return ret.next;       
    }
}
